master/include/logic/checkupManager.hpp

```diff
@@ -68,75 +94,179 @@ class CheckupManager {
+            // Watchdog_is_ready == 0
+            if (initialCheckupTimestamp.check() && !_systemData->digitalData.watchdog_state) {
+                state = CheckupState::CLOSE_SDC;
+                //Start toggling watchdog again (will only actually happen when the car is in AS_READY)
+            }
+            break;
+        case CheckupState::CLOSE_SDC:
+            // Close SDC
+            digitalWrite(SDC_LOGIC_CLOSE_SDC_PIN, LOW);
+            digitalWrite(MASTER_SDC_OUT_PIN, LOW);
+            state = CheckupState::WAIT_FOR_TS;
+            break;
+        case CheckupState::WAIT_FOR_TS:
+            // TS Activated?
+            if (_systemData->digitalData.aats_on) {
+                state = CheckupState::TOGGLE_VALVE;
+            }
+            break;
+        case CheckupState::TOGGLE_VALVE:
+            // Toggle EBS Valves
+            digitalWrite(EBS_VALVE_1_PIN, HIGH);
+            digitalWrite(EBS_VALVE_2_PIN, HIGH);
+
+            initialCheckupTimestamp.reset();
+            state = CheckupState::CHECK_PRESSURE;
+            break;
+        case CheckupState::CHECK_PRESSURE: {
+            // Check hyraulic line pressure and pneumatic line pressure
+            if (initialCheckupTimestamp.check()) {
+                return CheckupError::ERROR;
+            }
+            if (_systemData->sensors._hydraulic_line_pressure > 0 && _systemData->digitalData.
+                pneumatic_line_pressure) {
+                state = CheckupState::CHECK_TIMESTAMPS;
+            }
+            break;
+        }
+        case CheckupState::CHECK_TIMESTAMPS:
+            // Check if all components have responded and no emergency signal has been sent
+            if (_systemData->failureDetection.hasAnyComponentTimedOut() || _systemData->failureDetection.
+                emergencySignal) {
+                return CheckupError::ERROR;
+            }
+            return CheckupError::SUCCESS;
+        default:
+            break;
+    }
+    return CheckupError::WAITING_FOR_RESPONSE;
+}
+
+inline bool CheckupManager::shouldRevertToOffFromReady() const {
+    //TODO: UPDATE BRAKE PRESSURE CONDITION
+    if (!_systemData->digitalData.asms_on || !_systemData->digitalData.aats_on || _systemData->sensors.
+        _hydraulic_line_pressure == 0) {
+        return true;
+    }
+    return false;
 }
 
-inline bool CheckupManager::r2dCheckup() const {
+inline bool CheckupManager::shouldStayR2D() const {
     if (!_systemData->internalLogics.goSignal) {
-        return EXIT_FAILURE;
+        return true;
     }
-    return EXIT_SUCCESS;
+
+    _systemData->digitalData.watchdogTimestamp.reset();
+    return false;
 }
 
-inline bool CheckupManager::emergencyCheckup() const {
-    //TODO Continuous monitoring sequence
-    return EXIT_SUCCESS;
+inline bool CheckupManager::shouldEnterEmergency() const {
+    if (_systemData->failureDetection.hasAnyComponentTimedOut() ||
+        _systemData->failureDetection.emergencySignal ||
```

Comment:
Missing check to pneumatic sensors
```suggestion
        _systemData->digitalData.pneumatic_line_pressure || 
```